packages/Webkul/Shop/tests/Concerns/ShopTestBench.php

```diff
@@ -18,4 +28,482 @@ public function loginAsCustomer(?CustomerContract $customer = null): CustomerCon
 
         return $customer;
     }
+
+    /**
+     * Assert the cart.
+     */
+    public function assertCart(Cart $cart): void
+    {
+        $cart->refresh();
+
+        $this->assertModelWise([
+            Cart::class => [
+                [
+                    'id'                    => $cart->id,
+                    'customer_email'        => $cart->customer_email,
+                    'customer_first_name'   => $cart->customer_first_name,
+                    'customer_last_name'    => $cart->customer_last_name,
+                    'shipping_method'       => $cart->shipping_method,
+                    'coupon_code'           => $cart->coupon_code,
+                    'is_gift'               => $cart->is_gift,
+                    'items_count'           => $cart->items_count,
+                    'items_qty'             => $cart->items_qty,
+                    'exchange_rate'         => $cart->exchange_rate,
+                    'global_currency_code'  => $cart->global_currency_code,
+                    'base_currency_code'    => $cart->base_currency_code,
+                    'channel_currency_code' => $cart->channel_currency_code,
+                    'cart_currency_code'    => $cart->cart_currency_code,
+                    'grand_total'           => $cart->grand_total,
+                    'base_grand_total'      => $cart->base_grand_total,
+                    'sub_total'             => $cart->sub_total,
+                    'base_sub_total'        => $cart->base_sub_total,
+                    'tax_total'             => $cart->tax_total,
+                    'base_tax_total'        => $cart->base_tax_total,
+                    'discount_amount'       => $cart->discount_amount,
+                    'base_discount_amount'  => $cart->base_discount_amount,
+                    'checkout_method'       => $cart->checkout_method,
+                    'is_guest'              => $cart->is_guest,
+                    'is_active'             => $cart->is_active,
+                    'applied_cart_rule_ids' => $cart->applied_cart_rule_ids,
+                    'customer_id'           => $cart->customer_id,
+                    'channel_id'            => $cart->channel_id,
+                ],
+            ],
+        ]);
+    }
```

Comment:
It should be like this,

~~~php
/**
 * Prepare the cart for assertion.
 */
public function prepareCart(Cart $cart): array
{
    return [
        'id'                    => $cart->id,
        'customer_email'        => $cart->customer_email,
        'customer_first_name'   => $cart->customer_first_name,
        'customer_last_name'    => $cart->customer_last_name,
        'shipping_method'       => $cart->shipping_method,
        'coupon_code'           => $cart->coupon_code,
        'is_gift'               => $cart->is_gift,
        'items_count'           => $cart->items_count,
        'items_qty'             => $cart->items_qty,
        'exchange_rate'         => $cart->exchange_rate,
        'global_currency_code'  => $cart->global_currency_code,
        'base_currency_code'    => $cart->base_currency_code,
        'channel_currency_code' => $cart->channel_currency_code,
        'cart_currency_code'    => $cart->cart_currency_code,
        'grand_total'           => $cart->grand_total,
        'base_grand_total'      => $cart->base_grand_total,
        'sub_total'             => $cart->sub_total,
        'base_sub_total'        => $cart->base_sub_total,
        'tax_total'             => $cart->tax_total,
        'base_tax_total'        => $cart->base_tax_total,
        'discount_amount'       => $cart->discount_amount,
        'base_discount_amount'  => $cart->base_discount_amount,
        'checkout_method'       => $cart->checkout_method,
        'is_guest'              => $cart->is_guest,
        'is_active'             => $cart->is_active,
        'applied_cart_rule_ids' => $cart->applied_cart_rule_ids,
        'customer_id'           => $cart->customer_id,
        'channel_id'            => $cart->channel_id,
    ];
}

...

$cart->refresh();

$this->assertModelWise([
    Cart::class => [
        $this->prepareCart($cart),
    ],
]);
~~~

Reply:
done.